### DISCUSSION TITLE
Find connected nodes & edges from specific node index

### DISCUSSION THREAD

**Discussion by v-ngangarapu** [pyg-team/pytorch_geometric#8181] [Q&A]
Hi,
 I have trained a GNN model with multiple graphs that can predict node and edges. Using the trained model, I like to find in a graph, how the nodes  and edges are connected from a specific node. I see some examples with CaptumExplainer that is able find the connected nodes and edges and ploting it. How I can get list of connected edges & nodes so that I can use it for further analysis?

In the screen shot provided, the explainer able to identify 476, 10 & 2545 as connected, remaining as masked and one node is isolated. I want list of connected nodes and edges only and ignore remaining.

Any suggestions here.

![image](https://github.com/pyg-team/pytorch_geometric/assets/82965480/3a152631-26b4-4a62-b239-848f5b644fb6)

**Comment by wsad1** [comment]
The explainer returns and `edge_mask` and `node_mask` you can use this to obtain a list of edges/nodes that explain the prediction. 
See [here](https://pytorch-geometric.readthedocs.io/en/latest/tutorial/explain.html#explaining-node-classification-on-a-homogeneous-graph) for an example.

**Comment by v-ngangarapu** [comment]
Thanks for directing me to the tutorials. It was helpful. I understand based on intesity of weights in mask, the plotting of edges are getting varied. But I see similar weights for different node index within the graph. It's causing issue to filter the connected nodes.

code I am using:

explainer = Explainer(
    model=model,
    algorithm=CaptumExplainer('IntegratedGradients'),
    explanation_type='model',
    node_mask_type='attributes',
    edge_mask_type='object',
    model_config=dict(
        mode='multiclass_classification',
        task_level='node',
        return_type='log_probs',
    ),
)

node_index=1
explanation = explainer(dataset.data.x, dataset.data.edge_index.T, index=node_index)

got this plot for node_index : 8
Here, I want to select nodes - 7, 4
![image](https://github.com/pyg-team/pytorch_geometric/assets/82965480/1d774ff5-e00b-4d61-9781-5e0a34935999)


got this plot for node_index : 1
Here, I want to select nodes - 8, 7, 4
![image](https://github.com/pyg-team/pytorch_geometric/assets/82965480/74d3670c-3eba-443c-b8f9-f4c6be2e4468)

**Reply by rusty1s** [reply]
Sorry for late reply. PyG is not a graph analytics library, so we don't provide such functionality. I would suggest you can convert your `Explanation` to `networkx` and find meaningful subgraph/connected components from there. Would that work for you?

**Reply by v-ngangarapu** [reply]
Sorry, I got busy  with other priorities and not able to reply to this.

For my research, want to solve this problem with neural network instead of networkx. I believe extending the PyG with this feature may open the door for several other use cases. I got one suggestion to apply probabilities on results, do you think any of this kind of logic will help here?

**Reply by rusty1s** [reply]
What do you mean by applying probabilities on results?

**Reply by v-ngangarapu** [reply]
It's same like measuring the weight of results. Similar to whatever explainer does. I was trying to get something with this approach. This was not helping me.
